express-notes-api/server.ts

```diff
@@ -0,0 +1,182 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+
+type Note = {
+  id: number;
+  content: string;
+};
+
+type Notes = Record<number, Note>;
+
+type Data = {
+  nextId: number;
+  notes: Notes;
+};
+
+async function readNotes() {
+  const jsonData = await readFile('./data.json', 'utf8');
+  const data: Data = JSON.parse(jsonData);
+  return data;
+}
+
+async function writeNotes(data: Data) {
+  const jsonUpdatedData = JSON.stringify(data, null, 2);
+  await writeFile('./data.json', jsonUpdatedData);
+}
+
+app.use(express.json());
+
+app.get('/api/notes', async (req, res) => {
+  try {
+    const data = await readNotes();
+    const { notes } = data;
+    const notesArray: Note[] = [];
+
+    for (const id in notes) {
+      notesArray.push(notes[id]);
+    }
+
+    res.status(200).json(notesArray);
+  } catch (error) {
+    console.error('Error Message:', error);
```

Comment:
You need to return a response to the client here. Otherwise, you'll just get the default response from Express, which won't be pretty.

This is true for all of your catch clauses.